src/backend/dx12/src/device.rs

```diff
@@ -2374,6 +2375,10 @@ impl d::Device<B> for Device {
         (swapchain, hal::Backbuffer::Images(images))
     }
 
+    fn destroy_swapchain(&self, _swapchain: w::Swapchain) {
+        // Just drop
```

Comment:
I don't think that's correct in the current implementation.
The swapchain gets drop due to ComPtr, I agree.
But as far as I know, the retrieved backbuffer images need to be also released.

A nice smoketest would probably to try implementing resize in quad.

Reply:
Do you mean modifying method's signature in `hal::Device` to accept both the swapchain and backbuffer, e.g. `fn destroy_swapchain(&self, swapchain: B::Swapchain);` -> `fn destroy_swapchain(&self, swapchain: B::Swapchain, backbuffer: Backbuffer<B>);`?

Reply:
No, the signature is fine.
Swapchain has associated presentable images (GetBackbufferImage internally). We provide access to these images via the backbuffer. Presentable images are not supposed to be destroyed via `destroy_image` but freed alongside the swapchain on `destroy_swapchain`. In D3D12 we need to keep track of these presentable images inside the swapchain on creation and release them on destruction else they would be dangling. I think the best way would be to just store references inside the swapchain struct.